miden-lib/asm/miden/contracts/faucets/basic_fungible.masm

```diff
@@ -82,7 +82,7 @@ export.distribute.1
     # => [note_idx, EMPTY_WORD, EMPTY_WORD, ...]
 
     # load the ASSET and add it to the note
-    padw loc_loadw.0 movup.4 exec.tx::add_asset_to_note
+    padw loc_loadw.0 exec.tx::add_asset_to_note dropw
     # => [note_idx, ASSET, EMPTY_WORD, ...]
```

Comment:
Why do we need to first do `padw` and then `dropw` here? Could we not do something like:
```
movdn.4 loc_loadw.0 movup.4 exec.tx::add_asset_to_note
```

Reply:
Hm, I think I made a mistake here, it slipped my mind that `tx::add_asset_to_note` have a `[ASSET, note_idx] -> [ASSET, note_idx]` stack transition, so we don't need to drop the word at the end. So something like this would work:
```text
movdn.4 loc_loadw.0 exec.tx::add_asset_to_note movup.4
```

It is a problem though that this procedure was returning a wrong result, and no test was failing, although we have tests for `distribute` procedure. I'll check why this happened.